base/regex.jl

```diff
@@ -328,26 +328,58 @@ findfirst(r::Regex, s::AbstractString) = findnext(r,s,firstindex(s))
 
 
 """
-    findall(pattern::Union{AbstractString,Regex}, string::AbstractString; overlap::Bool=false)
```

Comment:
Why change this? IMO looks much better as-is.

Reply:
It's too long and past the recommended line length.

Reply:
Can we at least change it to a non-weird style then? :grimacing: e.g.
```julia

findall(pattern::Union{AbstractString,Regex},
        string::AbstractString;
        overlap::Bool=false)
```

Reply:
How do you feel about reserving a blank line to separate the kwargs from the positional args:
```julia
findall(pattern::Union{AbstractString,Regex},
        string::AbstractString,
        ;
        overlap::Bool=false)
```

Reply:
Me? Not amazing

Reply:
Let's not use this PR to debate every person's individual preferred formatting. I've been using this style in Julia since forever so it has as much of a claim to being standard as any.

Reply:
My point is mainly that most functions are not formatted this way in the docstrings, so it looks inconsistent.